src/main/java/com/nekomaster1000/infernalexp/items/GlowsilkBowItem.java

```diff
@@ -0,0 +1,98 @@
+
+	@Override
+	public void onPlayerStoppedUsing(ItemStack stack, World worldIn, LivingEntity entityLiving, int timeLeft) {
+		if (entityLiving instanceof PlayerEntity) {
+			PlayerEntity playerentity = (PlayerEntity)entityLiving;
+			boolean flag = playerentity.abilities.isCreativeMode || EnchantmentHelper.getEnchantmentLevel(Enchantments.INFINITY, stack) > 0;
+			ItemStack itemstack = playerentity.findAmmo(stack);
+
+			int i = this.getUseDuration(stack) - timeLeft;
+			i = net.minecraftforge.event.ForgeEventFactory.onArrowLoose(stack, worldIn, playerentity, i, !itemstack.isEmpty() || flag);
+			if (i < 0) return;
+
+			if (!itemstack.isEmpty() || flag) {
+				if (itemstack.isEmpty()) {
+					itemstack = new ItemStack(Items.ARROW);
+				}
+
+				float f = getArrowVelocity(i);
+				if (!((double)f < 0.1D)) {
+					boolean flag1 = playerentity.abilities.isCreativeMode || (itemstack.getItem() instanceof ArrowItem && ((ArrowItem)itemstack.getItem()).isInfinite(itemstack, stack, playerentity));
+					if (!worldIn.isRemote) {
+						ArrowItem arrowitem = (ArrowItem)(itemstack.getItem() instanceof ArrowItem ? itemstack.getItem() : Items.ARROW);
+						AbstractArrowEntity abstractarrowentity = arrowitem.createArrow(worldIn, itemstack, playerentity);
+						abstractarrowentity = customArrow(abstractarrowentity);
+						abstractarrowentity.setDirectionAndMovement(playerentity, playerentity.rotationPitch, playerentity.rotationYaw, 0.0F, f * 3.0F, 1.0F);
+						if (f == 1.0F) {
+							abstractarrowentity.setIsCritical(true);
+						}
+
+						int j = EnchantmentHelper.getEnchantmentLevel(Enchantments.POWER, stack);
+						if (j > 0) {
+							abstractarrowentity.setDamage(abstractarrowentity.getDamage() + (double)j * 0.5D + 0.5D);
+						}
+
+						int k = EnchantmentHelper.getEnchantmentLevel(Enchantments.PUNCH, stack);
+						if (k > 0) {
+							abstractarrowentity.setKnockbackStrength(k);
+						}
+
+						if (EnchantmentHelper.getEnchantmentLevel(Enchantments.FLAME, stack) > 0) {
+							abstractarrowentity.setFire(100);
+						}
+
+						stack.damageItem(1, playerentity, (p_220009_1_) -> {
+							p_220009_1_.sendBreakAnimation(playerentity.getActiveHand());
+						});
+						if (flag1 || playerentity.abilities.isCreativeMode && (itemstack.getItem() == Items.SPECTRAL_ARROW || itemstack.getItem() == Items.TIPPED_ARROW)) {
+							abstractarrowentity.pickupStatus = AbstractArrowEntity.PickupStatus.CREATIVE_ONLY;
+						}
+
+						if (abstractarrowentity instanceof ArrowEntity) {
+							((ArrowEntity) abstractarrowentity).addEffect(new EffectInstance(IEEffects.LUMINOUS.get(), 3600));
+						}
+
+						((AbstractArrowEntityAccess) abstractarrowentity).setGlow(true);
+
+						worldIn.addEntity(abstractarrowentity);
+					}
+
+					worldIn.playSound((PlayerEntity)null, playerentity.getPosX(), playerentity.getPosY(), playerentity.getPosZ(), SoundEvents.ENTITY_ARROW_SHOOT, SoundCategory.PLAYERS, 1.0F, 1.0F / (random.nextFloat() * 0.4F + 1.2F) + f * 0.5F);
```

Comment:
```suggestion
					worldIn.playSound(null, playerentity.getPosX(), playerentity.getPosY(), playerentity.getPosZ(), SoundEvents.ENTITY_ARROW_SHOOT, SoundCategory.PLAYERS, 1.0F, 1.0F / (random.nextFloat() * 0.4F + 1.2F) + f * 0.5F);
```
I'm pretty sure you don't need the typecast

Reply:
Oh yeah, probably not. Just copied that over from BowItem, so I didn't look to closely at all of it. Should we test it before making that change or assume it's fine?

Reply:
Nevermind, IntelliJ marks it as a redundant cast. Should be fine. Committing your suggestion and then merging!

Reply:
I believe World.playSound doesn't have any overloads that could be ambiguous to the type of the first parameter, so casting the null is unnecessary. You can commit this without worrying